Penteract/Assets/Scripts/DialogueManager.cpp

```diff
@@ -0,0 +1,308 @@
+	tutorialUpgrades2 = GameplaySystems::GetGameObject(tutorialUpgrades2UID);
+	tutorialUpgrades3 = GameplaySystems::GetGameObject(tutorialUpgrades3UID);
+
+	if (fangTextObject && onimaruTextObject) {
+		fangTextComponent = fangTextObject->GetComponent<ComponentText>();
+		onimaruTextComponent = onimaruTextObject->GetComponent<ComponentText>();
+	}
+
+	// UPGRADES
+	dialoguesArray[0] = Dialogue(2, "Hey Fang, look at this.", &dialoguesArray[1]);
+	dialoguesArray[1] = Dialogue(1, "Hmm...\nIt looks like Milibot has been\nresearching in some\nnew technologies...", &dialoguesArray[2]);
+	dialoguesArray[2] = Dialogue(1, "I might be able to\nseize its power for ourselves\nif we find a couple more.", &dialoguesArray[3]);
+	dialoguesArray[3] = Dialogue(8, "", nullptr);
+
+	dialoguesArray[4] = Dialogue(9, "", nullptr);
+
+	
+	dialoguesArray[5] = Dialogue(1, "I think I got it...\nYou can power it up this way,\nthen connect this here...\nand that there...", &dialoguesArray[6]);
+	dialoguesArray[6] = Dialogue(1, "Plug it in our core and...\nWHOAH! Oni, try this!", &dialoguesArray[7]);
+	dialoguesArray[7] = Dialogue(2, "I am not sure about this Fang...\nBut OK, I trust you.", &dialoguesArray[8]);
+	dialoguesArray[8] = Dialogue(10, "", nullptr);
+
+	// SWAP DIALOGUE + ONIMARU TUTORIAL
+	dialoguesArray[9] = Dialogue(1, "Oni, you perform better\nin closed spaces.\nTake the lead here!", &dialoguesArray[10]);
+	dialoguesArray[10] = Dialogue(7, "", &dialoguesArray[11]);
+	dialoguesArray[11] = Dialogue(2, "OK Fang.\nWatch how it is done.", &dialoguesArray[12]);
+	dialoguesArray[12] = Dialogue(6, "", nullptr);
+
+}
+
+void DialogueManager::Update() {
+	if (!fangTextComponent) return;
+	if (!onimaruTextComponent) return;
+	if (!tutorialFang) return;
+	if (!tutorialOnimaru) return;
+	if (!tutorialSwap) return;
+	if (!tutorialUpgrades1) return;
+	if (!tutorialUpgrades2) return;
+	if (!tutorialUpgrades3) return;
+
+	if (activeDialogue) {
+		if (runOpenAnimation) ActivateDialogue(activeDialogue);
+
+		if (Input::GetKeyCodeDown(Input::KEY_F) && !(runOpenAnimation || runChangeAnimation || runCloseAnimation) && activeDialogueObject) {
+			if (activeDialogue->nextDialogue) {
+				runChangeAnimation = true;
+				if ((activeDialogue->character < 5 && activeDialogue->nextDialogue->character >= 5) ||
+					(activeDialogue->character >= 5 && activeDialogue->nextDialogue->character < 5)) {
+					runCloseAnimation = true;
+					runSecondaryOpen = true;
+				}
+			}
+			else runCloseAnimation = true;
+		}
+
+		if (runChangeAnimation && !runCloseAnimation) {
+			ActivateNextDialogue(activeDialogue);
+		}
+
+		if (runCloseAnimation) CloseDialogue(activeDialogue);
+	}
+}
+
+void DialogueManager::SetActiveDialogue(Dialogue* dialogue, bool runAnimation)
+{
+	if (activeDialogueObject) {
+		if (activeDialogueObject->IsActive()) {
+			activeDialogueObject->Disable();
+		}
+	}
+
+	if (dialogue) {
+		if (dialogue->character >= 5) {
```

Comment:
Don't use magic numbers, use constants or defines instead so you don't have to guess what that magic number is